tests/unit/mathematics/linalg/operations/Viennacl_operations_unittest.cc

```diff
@@ -28,12 +28,39 @@ TEST(LinalgBackendViennaCL, add)
 	A_gpu = to_gpu(A);
 	B_gpu = to_gpu(B);
 
-	auto result = add(A, B, alpha, beta);
+	auto result_gpu = add(A_gpu, B_gpu, alpha, beta);
+	auto result = from_gpu(result_gpu);
 
 	for (index_t i = 0; i < 9; ++i)
 		EXPECT_NEAR(alpha*A[i]+beta*B[i], result[i], 1e-15);
 }
 
+TEST(LinalgBackendViennaCL, add_in_place)
+{
+	sg_linalg->set_gpu_backend(new LinalgBackendViennaCL());
+
+	const float32_t alpha = 0.3;
+	const float32_t beta = -1.5;
+
+	SGVector<float32_t> A(9), B(9), C(9);
+	SGVector<float32_t> A_gpu, B_gpu;
+
+	for (index_t i = 0; i < 9; ++i)
+	{
+		A[i] = i;
+		B[i] = 0.5*i;
+		C[i] = i;
+	}
+	A_gpu = to_gpu(A);
+	B_gpu = to_gpu(B);
+
+	add(A_gpu, B_gpu, A_gpu, alpha, beta);
```

Comment:
@OXPHOS the 'necessity' of allocating the memory advance is because some people want to actually have full control of the mallocs happening within the code